sys/usb/usbus/cdc/acm/cdc_acm_stdio.c

```diff
@@ -75,8 +93,19 @@ static void _cdc_acm_rx_pipe(usbus_cdcacm_device_t *cdcacm,
                              uint8_t *data, size_t len)
 {
     (void)cdcacm;
+#if CONFIG_USBUS_CDC_ACM_STDIO_BLOCKING
+    thread = thread_getpid();
+#endif
+
     for (size_t i = 0; i < len; i++) {
+#if CONFIG_USBUS_CDC_ACM_STDIO_BLOCKING
+        while (isrpipe_write_one(&_cdc_stdio_isrpipe, data[i]) < 0) {
+            xtimer_set(&timeout, XTIMER_BACKOFF+1); // shortest possible non-spinning wait
+            thread_sleep();
```

Comment:
how is this different from

```C
xtimer_sleep(XTIMER_BACKOFF + 1);
```

Reply:
Probably at all, if you say so :-)
I've tried xtimer_usleep but it was getting stuck, so I started digging further and came up with the awkward solution you see. I just did not realize it was only me waiting for too short a time, likely - I'll test againg.
Thanks for the tip!

Reply:
With your suggestion I've come up with a **much** cleaner solution. Thank you so much!